src/dreyfus_index.erl

```diff
@@ -120,23 +120,34 @@ init({DbName, Index}) ->
 
 handle_call({await, RequestSeq}, From,
             #state{
-                index=#index{current_seq=Seq}=Index,
+                index=#index{dbname=DbName,name=IdxName,ddoc_id=DDocId,current_seq=Seq}=Index,
                 index_pid=IndexPid,
                 updater_pid=nil,
                 waiting_list=WaitList
             }=State) when RequestSeq > Seq ->
-    UpPid = spawn_link(fun() -> dreyfus_index_updater:update(IndexPid, Index) end),
-    {noreply, State#state{
-        updater_pid=UpPid,
-        waiting_list=[{From,RequestSeq}|WaitList]
-    }};
+    DbName2 = mem3:dbname(DbName),
+    <<"_design/", GroupId/binary>> = DDocId,
+    NewState = case dreyfus_util:in_black_list(DbName2, GroupId, IdxName) of
+        false ->
+            UpPid = spawn_link(fun() ->
+                dreyfus_index_updater:update(IndexPid,Index)
+            end),
+            State#state{
+                updater_pid=UpPid,
+                waiting_list=[{From,RequestSeq}|WaitList]
+            };
+        _ ->
+            couch_log:notice("Index Blocked from Updating - db: ~p, ddocid: ~p name: ~p", [DbName, DDocId, IdxName]),
+            State
+    end,
+    {noreply, NewState};
 handle_call({await, RequestSeq}, _From,
             #state{index=#index{current_seq=Seq}}=State) when RequestSeq =< Seq ->
     {reply, {ok, State#state.index_pid, Seq}, State};
 handle_call({await, RequestSeq}, From, #state{waiting_list=WaitList}=State) ->
-    {noreply, State#state{
```

Comment:
why `noreply` is removed?

Reply:
it's at the bottom

Reply:
The diff for this clause looks: 
```
 handle_call({await, RequestSeq}, From, #state{waiting_list=WaitList}=State) ->
-    {noreply, State#state{
+    State#state{
         waiting_list=[{From,RequestSeq}|WaitList]
-    }};
+    };
```
Please remind me where is `noreply`.

Reply:
@jiangphcn look at the entire file, line 144.

Reply:
@jiangphcn : nevermind, not sure why that was changed, okay I see it now